### PR TITLE
Mariuszhussariush/ch7763/create embeddable additional cost calculator calculations

### PR DESCRIPTION
**Pull request by mariusz-hussariush**
added calculation methods, added bar support,  added form validation, added util methods

### CODE REVIEW

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx [hunk 1]
```diff
@@ -18,40 +20,44 @@ export class MyComponent {
...
-    this[name] = value;
+    const {name, value, validity: {valid}} = event.target;
+    if(valid) {this[name] = value};
   }
   private handleChangeSelect(event): void {
-    const {value} = event.target;
-    this.land = value !== 'reset' ? value : undefined;
+    const { value, validity: { valid }} = event.target;
+    if(valid) {
+      this.land = value !== 'reset' ? value : undefined;
```
Comment (hariscustovic) [inline additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: not null? 😛

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx [hunk 2]
```diff
@@ -68,18 +74,18 @@ export class MyComponent {
         </div>
         <div>
           <label htmlFor="brokerProvisonInput">
-            <input checked={this.withBrokerProvision} type="checkbox" id="brokerProvision" name="withBrokerProvision" onChange={(event) => this.handleChangeCheckbox(event)}/>
-            Mit Maklerprovision
-          </label>{this.withBrokerProvision}
-          <input value={this.brokerProvision} type="number" step="0.01" name="brokerProvision" placeholder="3.75 %" disabled={!!!this.withBrokerProvision} onChange={(event) => this.handleChange(event)}/>
+              <input checked={this.withBrokerCommission} type="checkbox" id="brokerProvisonInput" name="withBrokerCommission" onChange={(event) => this.handleChangeCheckbox(event)}/>
+            Mit MaklerComission
```
Comment (hariscustovic) [inline additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: as far as I know, Commission is not a German word. Should this not be Provision?

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx [hunk 3]
```diff
@@ -68,18 +74,18 @@ export class MyComponent {
         </div>
         <div>
           <label htmlFor="brokerProvisonInput">
-            <input checked={this.withBrokerProvision} type="checkbox" id="brokerProvision" name="withBrokerProvision" onChange={(event) => this.handleChangeCheckbox(event)}/>
-            Mit Maklerprovision
-          </label>{this.withBrokerProvision}
-          <input value={this.brokerProvision} type="number" step="0.01" name="brokerProvision" placeholder="3.75 %" disabled={!!!this.withBrokerProvision} onChange={(event) => this.handleChange(event)}/>
+              <input checked={this.withBrokerCommission} type="checkbox" id="brokerProvisonInput" name="withBrokerCommission" onChange={(event) => this.handleChangeCheckbox(event)}/>
```
Comment (hariscustovic) [inline additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: I would maybe change the ID to "brokerCommissionInput", just to stay consistent.

**Comment by kriim** [comment]
Do we want to give `my-component` a more meaningfull name?

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx (hunk 2 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: true

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx (hunk 3 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: done

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx (hunk 1 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: null indeed

**Pull request merged**